src/server/session.ts

```diff
@@ -2479,19 +2554,26 @@ export class Session<TMessage = string> implements EventSender {
         }
     }
 
-    private toPendingErrorCheck(uncheckedFileName: string): PendingErrorCheck | undefined {
-        const fileName = toNormalizedPath(uncheckedFileName);
-        const project = this.projectService.tryGetDefaultProjectForFile(fileName);
-        return project && { fileName, project };
-    }
-
-    private getDiagnostics(next: NextStep, delay: number, fileNames: string[]): void {
+    private getDiagnostics(next: NextStep, delay: number, fileArgs: (string | protocol.FileRangesRequestArgs)[]): void {
         if (this.suppressDiagnosticEvents) {
             return;
         }
 
-        if (fileNames.length > 0) {
-            this.updateErrorCheck(next, fileNames, delay);
+        if (fileArgs.length > 0) {
+            const files = mapDefined(fileArgs.filter(isString), file => ({ fileName: toNormalizedPath(file) }));
+            const filesWithRange = mapDefined(fileArgs.filter((arg): arg is protocol.FileRangesRequestArgs => !isString(arg)), arg => {
+                const fileName = toNormalizedPath(arg.file);
+                const scriptInfo = this.projectService.getScriptInfo(fileName);
+                if (!scriptInfo) {
+                    return undefined;
+                }
+                const ranges = arg.ranges.map(range => this.getRange({ file: arg.file, ...range }, scriptInfo));
+                return {
+                    fileName,
+                    ranges,
+                };
+            });
+            this.updateErrorCheck(next, [...filesWithRange, ...files], delay);

```

Comment:
I am not 100% sold on we modifying the order. Why is it on us? why is it not on client? 
I would much prefer us just going in order like you had it before. 
Checking scriptInfo here is not ideal because we dont want to check for scriptinfo, its project and updates before we actually start calculating diagnostics. (project could change in between in background or new open request arrives etc) so this sorting functionality seems unnecessary and should not be our responsibility